### PR TITLE
FEAT: Enabled argocd as integration with devtron for GitOps

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description
Argocd module integration from devtron stack manager.

Fixes  https://github.com/devtron-labs/devtron/issues/2293

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactoring 

# How Has This Been Tested?
1)  fresh installation -> Base - after our release -> no modules come as installed  (then create helm release, then install cicd, then create devtron-app and helm release, then install argo-cd, then create devtron-app and helm release)
2) fresh installation -> Base + cicd -> cicd module should come as installed. then create devtron-app and helm release, then install argo-cd, then create devtron-app and helm release 
3) fresh installation -> Base + cicd + argo-cd-> cicd and argo-cd modules come as installed, then create devtron-app and helm release (before configuring gitops and after)
4) fresh installation -> Base + argo-cd  -> is it allowed?  -> Pawan (drive from documentation)
5) I have already installed 5.0 version of devtron.
		5.1) I installed full mode -> after upgrading to our release, both cicd and argo-cd modules should come as installed (infra and code)
		5.2) I installed ea_only mode and I had installed cicd from UI -> after upgrading to our release, both cicd and argo-cd modules should come as installed (infra and code)
		5.3) I installed ea_only mode -> after upgrading to our release, both infra should be hyperion and both cicd and argo-cd should come as not-installed
6) I am enterprise user and on 0.5.0 version of devtron. now I am upgrading stack to 0.5.1 using pipeline -> both cicd and argo-cd modules should come as installed (infra and code)
7) programatically server update
8) think for security integration as well. 

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pkg/app/AppService.go [hunk 1]
```diff
@@ -591,7 +591,8 @@ func (impl AppServiceImpl) releasePipeline(pipeline *pipelineConfig.Pipeline, ar
 }
 
 func (impl AppServiceImpl) buildACDContext() (acdContext context.Context, err error) {
-	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken()
+	//this method should only call in case of argo-integration and gitops configured
+	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken("EMPTY-TOKEN")
```
Comment (manish-agrawal-ai) [inline pkg/app/AppService.go:595]: send empty.

File: pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go [hunk 2]
```diff
@@ -319,7 +319,8 @@ func (impl AppStoreDeploymentFullModeServiceImpl) createInArgo(chartGitAttribute
 
 func (impl AppStoreDeploymentFullModeServiceImpl) GetGitOpsRepoName(appName string, environmentName string) (string, error) {
 	gitOpsRepoName := ""
-	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken()
+	//this method should only call in case of argo-integration and gitops configured
+	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken("EMPTY-TOKEN")
```
Comment (manish-agrawal-ai) [inline pkg/appStore/deployment/fullMode/AppStoreDeploymentFullModeService.go:323]: send empty

File: pkg/pipeline/CdHandler.go [hunk 3]
```diff
@@ -209,8 +209,9 @@ func (impl *CdHandlerImpl) CheckArgoAppStatusPeriodicallyAndUpdateInDb(timeForDe
 }
 
 func (impl *CdHandlerImpl) GetAppStatusByResourceTreeFetchFromArgo(appName string) (timelineStatus pipelineConfig.TimelineStatus, appStatus, statusMessage string) {
+	//this should only be called when we have git-ops configured
 	//try fetching status from argo cd
-	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken()
+	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken("EMPTY-TOKEN")
```
Comment (manish-agrawal-ai) [inline pkg/pipeline/CdHandler.go:214]: EMPTY-TOKEN

File: pkg/pipeline/WorkflowDagExecutor.go [hunk 4]
```diff
@@ -1367,7 +1367,8 @@ func (impl *WorkflowDagExecutorImpl) subscribeHibernateBulkAction() error {
 }
 
 func (impl *WorkflowDagExecutorImpl) buildACDContext() (acdContext context.Context, err error) {
-	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken()
+	//this part only accessible for acd apps hibernation, if acd configured it will fetch latest acdToken, else it will return error
+	acdToken, err := impl.argoUserService.GetLatestDevtronArgoCdUserToken("EMPTY-TOKEN")
```
Comment (manish-agrawal-ai) [inline pkg/pipeline/WorkflowDagExecutor.go:1371]: EMPTY-TOKEN

File: pkg/util/TokenCache.go [hunk 5]
```diff
@@ -74,11 +73,11 @@ func GetACDAuthConfig() (*ACDAuthConfig, error) {
 	if err != nil {
 		return nil, err
 	}
-	if cfg.ACDPassword == "" {
+	/*if cfg.ACDPassword == "" {
```
Comment (manish-agrawal-ai) [inline pkg/util/TokenCache.go:76]: please add comment

**Review by manish-agrawal-ai — COMMENTED** [review]
check comment

File: util/argo/ArgoUserService.go [hunk 6]
```diff
@@ -78,6 +83,21 @@ func GetDevtronSecretName() (*DevtronSecretConfig, error) {
...
+	gitOpsConfig, err := impl.gitOpsRepository.GetGitOpsConfigActive()
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("GetGitOpsConfigActive, error while getting", "err", err)
+		return
+	}
+	if gitOpsConfig != nil && gitOpsConfig.Id > 0 {
+		isGitOpsConfigured = true
+	}
+	if !isGitOpsConfigured {
+		//TODO FIX - CHECK INTEGRATION AVAILABLE OR NOT
```
Comment (nishant-d) [inline util/argo/ArgoUserService.go:97]: do we need some impl here ?

**Review by nishant-d — CHANGES_REQUESTED** [review]

**Review by nishant-d — DISMISSED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**